lang/gcc13/Portfile

```diff
@@ -0,0 +1,561 @@
+
+epoch               0
+name                gcc13
+
+homepage            https://gcc.gnu.org/
+
+platforms           {darwin >= 10}
+categories          lang
+maintainers         nomaintainer
+# an exception in the license allows dependents to not be GPL
+license             {GPL-3+ Permissive}
+
+description         The GNU compiler collection
+long_description    {*}${description}, including front ends for \
+                    C, C++, Objective-C, Objective-C++ and Fortran. \
+                    NOTE : NOT YET READY FOR MAINSTREAM USE \
+                    This version is test version to validate what support is \
+                    like accross the various OS versions. It INTENTIONALLY is \
+                    currently in conflict with the gcc12 port.
+
+# Remember to reset all revision increments below to 0 on new versions
+version             13.2.0
+revision            0
+
+set libgccname      lib${name}
+subport             ${libgccname} { revision [ expr ${revision} + 0 ] }
+
+set libcxxname      ${name}-libcxx
+
+checksums           rmd160  a6d646ed9765f973d3f63ef560edf4a50cf686c3 \
+                    sha256  e275e76442a6067341a27f04c5c6b83d8613144004c0413528863dc6b5c743da \
+                    size    87858592
+
+# Primary releases
+master_sites        https://ftpmirror.gnu.org/gcc/gcc-${version}/ \
+                    https://mirror.its.dal.ca/gnu/gcc/gcc-${version}/ \
+                    https://mirrors.kernel.org/gnu/gcc/gcc-${version}/ \
+                    https://www.mirrorservice.org/sites/ftp.gnu.org/gnu/gcc/gcc-${version}/ \
+                    https://ftp-stud.hs-esslingen.de/pub/Mirrors/ftp.gnu.org/gcc/gcc-${version}/ \
+                    https://mirror.yongbok.net/gnu/gcc/gcc-${version}/ \
+                    http://mirror.koddos.net/gcc/releases/gcc-${version}/ \
+                    ftp://ftp.gwdg.de/pub/linux/gcc/releases/gcc-${version}/ \
+                    ftp://gcc.ftp.nluug.nl/mirror/languages/gcc/releases/gcc-${version}/ \
+                    ftp://gcc.gnu.org/pub/gcc/releases/gcc-${version}/ \
+                    gnu:gcc/gcc-${version}
+
+distname            gcc-${version}
+use_xz              yes
+
+depends_build-append \
+                    port:texinfo
+depends_lib-append  port:cctools \
+                    port:gmp \
+                    path:lib/pkgconfig/isl.pc:isl \
+                    port:ld64 \
+                    port:libiconv \
+                    port:libmpc \
+                    port:mpfr \
+                    port:zlib \
+                    port:zstd
+depends_run-append  port:gcc_select \
+                    path:share/doc/libgcc/README:libgcc
+
+depends_skip_archcheck-append gcc_select ld64 cctools
+license_noconflict  gmp mpfr ppl libmpc zlib
+
+set major           [lindex [split ${version} .-] 0]
+
+platform darwin {
+    configure.pre_args-append --build=${build_arch}-apple-darwin${os.major}
+}
+
+set gcc_configure_langs {c c++ objc obj-c++ lto fortran}
```

Comment:
@cjones051073 Why don’t we add Modula and Rust support? They build even for PowerPC – should build for newer systems. Rust passes most tests too.
https://trac.macports.org/ticket/67354

Reply:
Stuff like that can come later, if you are willing to submit the PRs. There is more basic stuff to get right first like getting the ports as is to build. I am having issues in my machine here, macOS13 which I suspect are related to the mess Xcode 15 has caused with its changes to the linker. I was hoping to see via this PR if things worked on older OSes any better but those builds seem to time out so likely will have to just merge this and see what happens in the build bots.

Reply:
B.t.w. If you have access to a machine running an os older than macOS13, and if you are willing please check out the branch here and see if it builds for you or not ?

Reply:
gcc rust currently passes 14% of the rustc test suite, and I believe is widely considered way too young to be usable.

https://github.com/Rust-GCC/Reporting/blob/main/2023-08-monthly-report.org

<!DOCTYPE html>

rustc testsuite passing tests | gccrs | 14.0% | 14.0%
-- | -- | -- | --


<br class="Apple-interchange-newline">

Reply:
@cjones051073 Sure, I will try building it, but on Oct 2. (No access to fast hardware until then.)

Reply:
> gcc rust currently passes 14% of the rustc test suite, and I believe is widely considered way too young to be usable.

To be clear, I referred to `gccrs` own test-suite. That is, a part of GCC test-suite.

Reply:
Thanks fir the offer, but if someone else doesn't offer before then I might just hit go on merge and see what the buildbots say ;)

Please not as it stands the port isn't really usable, as I have intentionally not made it the default runtime provider, and have made it conflict with libgcc12 so you will have to deactivate that first before test building this.  For now the only use this port has is to see how it builds... once we have it building on all OSes we need to support only then can the changes be made to allow it to be used mire generally.

Reply:
@cjones051073 I see no downsides to merging it prior to testing on old systems, since Macports at the moment does not use new GCCs on old systems anyway. And since it does not replace `gcc12`, my set-up won’t be affected either.

Let us see how this builds and works, we can deal with fixes for old macOS in a while.

Reply:
@catap Maybe you could try building it on some older Intel macOS?

Reply:
> > gcc rust currently passes 14% of the rustc test suite, and I believe is widely considered way too young to be usable.
> 
> To be clear, I referred to `gccrs` own test-suite. That is, a part of GCC test-suite.

Well OK, but we can't confuse that with gccrs being a rust compiler for general use.

Reply:
Well, I did not make such a claim, and enabling it in GCC won’t affect anything Rust-related in Macports. But to test it, it has to be available in the first place.

Reply:
I personally have no problem with just enabling it in the builds even if it isn't 100% ready yet, as as you say if nothing uses it that in itself will not cause any issues, and once there it will make testing it easier.

What will be required though is it doesn't break the gcc port builds.... But that can be tested once we have the current version merged and someone submits the PRs to enable the rust frontend.

Reply:
enabling a nascent, mostly nonfunctional compiler that is probably still years away from being actually useful for anything as part of the default install is pretty much just asking for trouble and tickets ...

Reply:
@kencu It is not used for anything in Macports. As long as it builds, there is nothing to cause any tickets. If it does not work as intended for compiling Rust code, it is an upstream problem, not ours.

Reply:
> enabling a nascent, mostly nonfunctional compiler that is probably still years away from being actually useful for anything as part of the default install is pretty much just asking for trouble and tickets ... 

Then we enable it via a variant, which isn't enabled by default but allows those that wish to test to easily enable it. That probably is the way to go here.